src/Networking/MQTT/mqtt_pal.h

```diff
@@ -26,11 +26,14 @@ typedef time_t mqtt_pal_time_t;
 extern uint32_t millis() noexcept;
 #define MQTT_PAL_TIME() (millis() / 1000)
 
-// Mutex - define to nothing; MQTT-C is used in a single thread context.
-typedef int mqtt_pal_mutex_t;
-#define MQTT_PAL_MUTEX_LOCK(x)
-#define MQTT_PAL_MUTEX_INIT(x)
-#define MQTT_PAL_MUTEX_UNLOCK(x)
+typedef struct
```

Comment:
Please confirm that we need to use a mutex, i.e. there can be more than one task accessing those parts of the system that the mutex guards. Also, are we certain that no deadlock can occur?

Reply:
The mutex is for guarding resources between (1) the main task when it calls `mqtt_publish` - when buffering the message for sending and (2) the network task when it calls `mqtt_sync` - when moving data from mqtt buffer to socket buffer.

As to whether deadlocks can occur,  one of the condition for it does not seem to be satisfied - circular wait. `mqtt_sync` and `mqtt_publish` only share 1 common mutex they are both attempting to lock; and `mqtt_publish` itself only tries to attempt to lock 1 mutex, breaking the possibility of a circular wait chain from forming.